web/src/toga_web/widgets/selection.py

```diff
@@ -0,0 +1,41 @@
+from toga_web.libs import create_proxy
+
+from .base import Widget
+
+
+class Selection(Widget):
+
+    def create(self):
+        self.native = self._create_native_widget("sl-select")
+        self.native.addEventListener("sl-change", create_proxy(self.dom_onchange))
+
+    def dom_onchange(self, event):
+        self.interface.on_change()
+
+    def clear(self):
+        while self.native.firstElementChild:
+            self.native.removeChild(self.native.firstElementChild)
+
+    def insert(self, index, item):
+        display_text = self.interface._title_for_item(item)
+        option = self._create_native_widget("sl-option")
+        option.value = display_text
+        option.textContent = display_text
```

Comment:
"value" and "text content" are different attributes for a reason - the text labels in the selection don't have to be unique. The approach you're using here for identifying the currently selected index will always find the first matching text label. 

If Shoelace doesn't provide a native index-based selection mechanism, then you can use the `value` field to track the index.